ambry-clustermap/src/main/java/com.github.ambry.clustermap/Partition.java

```diff
@@ -193,7 +193,12 @@ public boolean equals(Object o) {
 
     Partition partition = (Partition) o;
 
-    return id == partition.id;
+    if (id == null ^ partition.id == null){
```

Comment:
I think this check is unnecessary and can be omitted. `id` is supposed to be non-null, the construction of `Partition` will fail otherwise.